File: leetcode/rust-leetcode/src/1460.make-two-arrays-equal-by-reversing-subarrays.rs
impl Solution {
    pub fn can_be_equal(target: Vec<i32>, arr: Vec<i32>) -> bool {
        if target.len() != arr.len() {
            return false;
        }
        let mut counts = [0; 1001];
        for t in target {
            counts [t as usize] += 1;
        }
        for a in arr {
            if counts[a as usize] == 0 {
                return false;
            }
            counts[a as usize] -= 1;
        }
        true
    }
}

File: leetcode/go/1460.make-two-arrays-equal-by-reversing-subarrays.go
func canBeEqual(target []int, arr []int) bool {
    if len(target) != len(arr) {
        return false
    }
    counts := make([]int, 1001)
    for _, n := range arr {
        counts[n]++
    }
    for _, n := range target {
        if counts[n] == 0 {
            return false
        }
        counts[n]--
    }
    return true
}

